src/plugins/intellisense/intellisense.int.test.ts

```diff
@@ -0,0 +1,92 @@
+  testWithPage(
+    "Intellisense Autocomplete",
+    async (driver) => {
+      await driver.enablePlugin("intellisense");
+      const blankState = await driver.getState();
+
+      const testIdentSample = async (
+        typedPrefix: string,
+        latexPrefix: string,
+        varname: string,
+        subscriptname?: string
+      ) => {
+        let typedIdentifierSample = `${varname}`;
+
+        // generate expected latex
+        let expectedLatex = varname;
+        if (varname.length > 1) {
+          expectedLatex = `\\${varname}`;
+        }
+        if (subscriptname) {
+          expectedLatex += `_{${
+            subscriptname.length === 0 ? " " : subscriptname
+          }}`;
+        }
+        if (subscriptname !== undefined) {
+          typedIdentifierSample += `_${subscriptname}`;
+        }
+
+        // create reference expression
+        await driver.keyboard.type(`${typedIdentifierSample}=123`);
+        await driver.keyboard.press("Enter");
+
+        // calculate how many "segments" it has
+        let segmentsLength = 1;
+        if (subscriptname?.length === 0) {
+          segmentsLength = 2;
+        } else if (subscriptname && subscriptname.length > 0) {
+          segmentsLength = 2 + subscriptname.length;
+        }
+
+        // try autocompleting from the first i "parts" of the identifier sample
+        for (let i = 1; i < segmentsLength; i++) {
+          // go left j segments to validate it works at any position
+          for (let j = 0; j < i - 1; j++) {
+            // generate partial identifier string
+            let str = varname;
+            if (i > 1 && subscriptname !== undefined) {
+              str += `_${subscriptname.slice(0, i - 2)}`;
+            }
+
+            // type out str and go left
+            await driver.keyboard.type(typedPrefix + str);
+            for (let k = 0; k < j; k++) {
+              await driver.keyboard.press("ArrowLeft");
+            }
+
+            // try autocomplete
+            await driver.keyboard.press("Enter");
+            await driver.assertSelectedItemLatex(
+              latexPrefix + expectedLatex,
+              `Testing Identifier '${typedIdentifierSample}', autocompleting from '${str}', going left ${j} characters.`
+            );
+
+            // go to next expr
+            await driver.keyboard.press("Enter");
+          }
+        }
+      };
+
+      await driver.focusIndex(0);
+      await testIdentSample("1+", "1+", "A", "1a3");
+      await driver.setState(blankState);
+      await driver.focusIndex(0);
+      await testIdentSample("1+", "1+", "alpha", "4b6");
+      await driver.setState(blankState);
+      await driver.focusIndex(0);
+      await testIdentSample("1+", "1+", "beta", "7c9");
+      await driver.setState(blankState);
+      await driver.focusIndex(0);
+      await testIdentSample("1+", "1+", "B");
+
+      await driver.clean();
+      return clean;
+    },
+    20000
```

Comment:
Do you need the longer timeout?

Reply:
Yeah that might be a bit overkill, though the intellisense tests do tend to be right on the edge of going past the default 5s maximum.